### PR TITLE
Add warning message for a long queries

### PR DESCRIPTION
**Pull request by AlexRAV**
Add warning message mensioned in following issues:
https://github.com/gatsbyjs/gatsby/issues/23098
https://github.com/gatsbyjs/gatsby/issues/21825

### CODE REVIEW

**Comment by KyleAMathews** [comment]
Sweet thanks!

Maybe drop it to 15 seconds? Queries should generally take milliseconds or a second or two if hitting a remote API to finish so a lower number seems appropriate and high signal still.

**Comment by AlexRAV** [comment]
Done)

**Comment by AlexRAV** [comment]
I see that something went wrong, but can't see what exactly as I don't have permissions for Google Cloud Platform. Could you somehow show me logs?

File: packages/gatsby/src/query/query-runner.js [hunk 1]
```diff
@@ -25,7 +25,28 @@ type QueryJob = {
...
+  const graphql = (query, context) => {
+    // Check if query takes too long, print out warning
+    const promise = graphqlRunner.query(query, context)
+    let isPending = true
+
+    const timeoutId = setTimeout(() => {
+      if (isPending) {
+        const { pages } = store.getState()
+        const page = pages.get(queryJob.id)
+        console.warn(
```
Comment (vladar) [inline packages/gatsby/src/query/query-runner.js:37]:
```suggestion
        report.warn(
```

File: packages/gatsby/src/query/query-runner.js [hunk 2]
```diff
@@ -25,7 +25,28 @@ type QueryJob = {
...
+          `Query in the ${page.internalComponentName} component takes too long`
```
Comment (vladar) [inline packages/gatsby/src/query/query-runner.js:38]: Query runner also executes [static queries](https://www.gatsbyjs.org/docs/static-query/) (not just page queries). Maybe we can use `queryJob.componentPath` here instead? It should be universal for both page and static queries.

**Comment by AlexRAV** [comment]
Thank you for review!

File: packages/gatsby/src/query/query-runner.js (hunk 2 above)
Comment (AlexRAV) [reply packages/gatsby/src/query/query-runner.js]: Yeah, I think that `queryJob.componentPath` looks more suitable. Thanks.

**Comment by pieh** [comment]
1. Can we conditionally add more information if it is actually page query (query running taking a long time can be specific to particular page / context)?

We do this for errors - https://github.com/gatsbyjs/gatsby/blob/5cd1d617e343d7f914abc363101c19cf1232a44d/packages/gatsby/src/query/query-runner.js#L46-L49

StaticQuery won't have `path` or `context` so this should only apply page queries.

2. This have potential to spam console - should there be some check to see if same warning was already issued? (for example if the long running query is not specific to particular page, but rather to particular page template - we could potentially display same warning for every page and they would exactly the same as warning right now doesn't mention page path/url)

---
The 2 questions above are somewhat mutually exclusive

**Comment by AlexRAV** [comment]
Hi!

@pieh Done.
@vladar any info about cloud tests?

**Comment by vladar** [comment]
Published in `gatsby@2.20.29`

**Pull request merged**